### PR TITLE
Revoke NC token on 'logout' and 'set defaults'

### PR DESCRIPTION
**Pull request by piotrjurkiewicz**

### CODE REVIEW

File: nc/nc.go
Comment (bartoszWojciechO) [inline nc/nc.go:485]: Technically management goroutine could still be running at this point, because `Stop` just calls `cancelConnecting` and doesn't wait for any kind of confirmation. So at this point management loop could be attempting to fetch new creds, connecting to NC, etc. Previously it was not an issue because in any context, there would always be a big delay between Starting/Stopping the NC. I'm not sure if now it's completely safe, since `Revoke` is called right after `Stop`. Might not be a big deal, since it's just the dev

File: nc/nc.go
Comment (bartoszWojciechO) [inline nc/nc.go:491]: This entire block(line 490-510) should be encapsulated in `CredsFetcher`.

File: daemon/rpc_set_defaults.go
Comment (mariusSincovici) [inline daemon/rpc_set_defaults.go:33]: same here, isn't needed `!IsProdEnv`?

File: nc/nc.go
Comment (mariusSincovici) [inline nc/nc.go:498]: could you add also the value for `purgeSession` in the log?

File: nc/nc.go
Comment (mariusSincovici) [inline nc/nc.go:508]: for error wouldn't make sense to log the `Status`?

File: nc/nc.go
Comment (piotrjurkiewicz) [reply nc/nc.go]: Done.

File: core/core.go
Comment (piotrjurkiewicz) [reply core/core.go:465]: Good idea. Request will not execute when appUserID is empty.

File: nc/nc.go
Comment (piotrjurkiewicz) [reply nc/nc.go]: Done.

File: nc/nc.go
Comment (piotrjurkiewicz) [reply nc/nc.go]: Done.

File: daemon/rpc_set_defaults.go
Comment (piotrjurkiewicz) [reply daemon/rpc_set_defaults.go:33]: `IsDevEnv` is literaly defined as `!IsProdEnv`:
```go
func IsDevEnv(env string) bool {
	return !IsProdEnv(env)
}
```

File: daemon/rpc_logout.go
Comment (piotrjurkiewicz) [reply daemon/rpc_logout.go:48]: It looks like that not. `QA` constant is not used:
```go
const (
	// Development defines development environment
	Development Environment = "dev"
	// QA defines qa environment
	QA = "qa"
	// Production defines production environment
	Production = "prod"
	// Downloader modifies configs and servers jobs
	Downloader = "downloader"
)
```

File: core/core.go
Comment (piotrjurkiewicz) [reply core/core.go:458]: I don't know. That's the description an implementation suggestion I received from the API maintainer.

File: daemon/rpc_set_defaults.go
Comment (mariusSincovici) [reply daemon/rpc_set_defaults.go:33]: yes, ignore :smile:

**Review by mariusSincovici — APPROVED** [review]

File: nc/nc.go
Comment (piotrjurkiewicz) [reply nc/nc.go:484]: I checked what will happen when credentials will be revoked without stopping the loop. The manager will just try to fetch new credentials. So it does not look like can break anything. In our code stop is esured to be called before revoke. And even if the management loop will not be able to finish after stop call before calling revoke, it will finish when trying to get new creds.

**Review by bartoszWojciechO — APPROVED** [review]

**Pull request merged**
